rs/rust_canisters/random_traffic_test/src/main.rs

```diff
@@ -0,0 +1,236 @@
+use candid::{Decode, Encode};
+use dfn_core::api;
```

Comment:
one nit - we are working to remove dfn_core, so if it's possible to use ic_cdk for this, that would be much better.

Reply:
That depends in whether it allows the distinction between what I called a synchronous rejection and asynchronous rejection. A synchronous is when the system tells me immediately "out of memory" or "output queue full", whereas an asynchronous rejection comes from a reject response, possibly originating on a different subnet.

From looking at [CallResult](https://docs.rs/ic-cdk/latest/ic_cdk/api/call/type.CallResult.html), it looks like an asynchronous rejection since a reject response has a message and an error code. OTOH [call_with_callbacks sends](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/rust_canisters/dfn_core/src/api.rs?L345) sends out a call but also gives me an `i32` (a reject code) immediately.

When I use `await` on [call](https://docs.rs/ic-cdk/latest/ic_cdk/api/call/fn.call.html) it doesn't look like I can distinguish between the two. This is why I'm not using ic_cdk.

Reply:
From looking at the [code](https://docs.rs/ic-cdk/latest/src/ic_cdk/api/call.rs.html#100-126) (and the [documentation](https://docs.rs/ic-cdk/latest/ic_cdk/api/call/fn.call.html)), you must `await` the future in order for the call to be made at all,

You could instead use the [`ic0` crate](https://docs.rs/ic0/0.23.0/ic0/) directly. It's all `unsafe` functions, but you can freely copy-paste from the `ic-cdk` to get what you want.

Reply:
I can copy paste the code for dfn core, which is just a wrapper for `ic0`, but then I'm not sure why is dfn core even being removed? Seems to me like having wrappers for these calls makes total sense.

Reply:
Similar yes but unfortunately ic_cdk isn't offering a functionality I am relying on here (as far as I know), i.e. the synchronous rejections feedback. I'll leave it like this for now since all I can is copy paste the code from dfn_core but then we can also do that if is is phased out together with other canisters using it.